components/region_cache_memory_engine/src/engine.rs

```diff
@@ -475,61 +518,55 @@ impl Iterator for RegionCacheIterator {
 }
 
 #[derive(Clone, Debug)]
-pub struct RegionCacheSnapshot {
-    region_id: u64,
+pub struct RangeCacheSnapshot {
+    range: CacheRange,
```

Comment:
maybe we can wrap range, snapshot_ts, sequence_number into RangeCacheSnapshotMeta struct. 
And use the RangeCacheSnapshotMeta for snapshot management.